typed-racket-lib/typed-racket/rep/rep-utils.rkt

```diff
@@ -50,14 +49,22 @@
 (define (normalize-id id)
   (free-id-table-ref! id-table id id))
 
-(define (hash-id id)
-  (eq-hash-code (identifier-binding-symbol id)))
-
-(define (hash-name-ref name rec)
-  (if (identifier? name)
-      (hash-id name)
-      (rec name)))
 
+;; function for fetching the interned copies of Reps
+;; should be robust to GC occurring at any time (since we
+;; not only check for the interned weak-box, but also then
+;; check that it contains a non-#f value)
+(define (intern-table-ref! table val)
+  (define intern-box (hash-ref table val #f))
```

Comment:
This code isn't thread-safe, which isn't a problem currently but could be in the future.